woocommerce/woocommerce
# Add support for rendering HTML inside gateway and plugin descriptions.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #52709 

* Add support for sanitised HMTL to be displayed inside description fields. HTML is sometimes returned from a plugin or gateway description, so we should support this case.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the `payment-gateway-list-item.tsx` file and replace line 74 as follows:

```typescript
decodeEntities( "This is a line break:<br/>This is <b>bold</b> text and this is an <a href='https://google.com'>example link</a>." )
```

2. Run ` pnpm run --filter=@woocommerce/plugin-woocommerce watch:build`
3. Refresh the `WooCommerce > Settings > Payments` page. You should see the string above rendered as HTML markup in the description. The line breaks, bold text, and `<a>` link should all be correctly rendered.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
